server/base/src/main/java/org/apache/accumulo/server/metadata/iterators/SetEqualityIterator.java

```diff
@@ -83,14 +83,22 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
       int count = 0;
 
       while (source.hasTop()) {
-        byte[] ba = source.getTopKey().getColumnQualifierData().toArray();
-        dos.writeInt(ba.length);
-        dos.write(ba, 0, ba.length);
-
         if (includeValue) {
+          byte[] ba = source.getTopKey().getColumnQualifierData().toArray();
           byte[] valueData = source.getTopValue().get();
-          dos.writeInt(valueData.length);
-          dos.write(valueData, 0, valueData.length);
+
+          byte[] encodedData = encodeEntry(Map.entry(ba, valueData));
```

Comment:
I did not know that Map.entry existed.  Looks like it was added in java v9.  This is not something for this PR, but I think we could replace all usage of AbstractMap.Simple*Entry in the code with Map.entry() in another PR.  Seems like there are least 35 usages in the code.

```
 find . -name "*.java" | xargs grep -E "SimpleImmutableEntry|SimpleEntry" | wc
     35     163    5131

```